gcloud-java-dns/src/main/java/com/google/gcloud/testing/OptionParsersAndExtractors.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.testing;
+
+import com.google.api.services.dns.model.Change;
+import com.google.api.services.dns.model.ManagedZone;
+import com.google.api.services.dns.model.Project;
+import com.google.api.services.dns.model.ResourceRecordSet;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility helpers for LocalDnsHelper.
+ */
+class OptionParsersAndExtractors {
+
+  /**
+   * Makes a map of list options. Expects query to be only query part of the url (i.e., what follows
+   * the '?').
+   */
+  static Map<String, Object> parseListZonesOptions(String query) {
+    Map<String, Object> options = new HashMap<>();
+    if (query != null) {
+      String[] args = query.split("&");
+      for (String arg : args) {
+        String[] argEntry = arg.split("=");
+        switch (argEntry[0]) {
+          case "fields":
+            // List fields are in the form "managedZones(field1, field2, ...)"
+            String replaced = argEntry[1].replace("managedZones(", ",");
+            replaced = replaced.replace(")", ",");
+            // we will get empty strings, but it does not matter, they will be ignored
+            options.put(
+                "fields",
+                replaced.split(","));
+            break;
+          case "dnsName":
+            options.put("dnsName", argEntry[1]);
+            break;
+          case "nextPageToken":
+            options.put("nextPageToken", argEntry[1]);
+            break;
+          case "pageToken":
+            options.put("pageToken", argEntry[1]);
+            break;
+          case "maxResults":
+            // parsing to int is done while handling
+            options.put("maxResults", argEntry[1]);
+            break;
+          default:
+            break;
+        }
+      }
+    }
+    return options;
+  }
+
+  /**
+   * Makes a map of list options. Expects query to be only query part of the url (i.e., what follows
```

Comment:
"map" looks like a copy paste error?  I think the method-level javadoc is also probably unnecessary in this file since everything is relatively clear and package scope.